plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -152,21 +155,49 @@ func (c *MaxPDVolumeCountChecker) filterVolumes(volumes []api.Volume, namespace
 		} else if vol.PersistentVolumeClaim != nil {
 			pvcName := vol.PersistentVolumeClaim.ClaimName
 			if pvcName == "" {
-				return fmt.Errorf("PersistentVolumeClaim had no name: %q", pvcName)
+				// if the PVC does not have a name, log the error
+				// and count the PV towards the PV limit
+				// generate a random volume ID since its required for de-dup
+				glog.Errorf("PersistentVolumeClaim with no name found in namespace: %q", namespace)
+				source := rand.NewSource(time.Now().UnixNano())
+				generatedID := "noClaimName" + strconv.Itoa(rand.New(source).Intn(1000000))
+				filteredVolumes[generatedID] = true
+				return nil
 			}
 			pvc, err := c.pvcInfo.GetPersistentVolumeClaimInfo(namespace, pvcName)
 			if err != nil {
-				return err
+				// if the PVC is not found, log the error
+				// and count the PV towards the PV limit
+				// generate a random volume ID since its required for de-dup
+				glog.Error(err)
+				source := rand.NewSource(time.Now().UnixNano())
+				generatedID := "missingPVC" + strconv.Itoa(rand.New(source).Intn(1000000))
+				filteredVolumes[generatedID] = true
+				return nil
 			}
 
 			pvName := pvc.Spec.VolumeName
 			if pvName == "" {
-				return fmt.Errorf("PersistentVolumeClaim is not bound: %q", pvcName)
+				// if the PVC is not bound, log the error
+				// and count the PV towards the PV limit
+				// generate a random volume ID since its required for de-dup
+				glog.Errorf("PersistentVolumeClaim is not bound: %q", pvcName)
+				source := rand.NewSource(time.Now().UnixNano())
+				generatedID := "unboundPVC" + strconv.Itoa(rand.New(source).Intn(1000000))
+				filteredVolumes[generatedID] = true
+				return nil
```

Comment:
In case the PVC is not yet bound - we could fail at the pod scheduling stage or let the pod be scheduled and fail to be created on the node. In either case, I believe retries should result in the pod being successfully created (is this true?). However, I think it is a better user experience to let the pod fail scheduling with the appropriate error.

Also, since this is a functional change and not required to resolve the underlying issue, I am going to remove this change from this PR.